toolkit/tools/pkg/imagecustomizerlib/artifactsinputoutput.go

```diff
@@ -95,15 +144,143 @@ func outputArtifacts(items []imagecustomizerapi.OutputArtifactsItemType, outputD
 		destPath := filepath.Join(outputDir, bootConfig.systemdBootBinary)
 		err := file.Copy(srcPath, destPath)
 		if err != nil {
-			return fmt.Errorf("failed to copy binary from (%s) to (%s):\n%w", srcPath, destPath, err)
+			return nil, fmt.Errorf("failed to copy binary from (%s) to (%s):\n%w", srcPath, destPath, err)
 		}
+
+		signedPath := prefix + replaceSuffix(filepath.Join(configOutputArtifactsPath, bootConfig.systemdBootBinary), ".efi", ".signed.efi")
+
+		outputArtifactsMetadata = append(outputArtifactsMetadata, OutputArtifactMetadata{
+			Partition:   partition,
+			Source:      signedPath,
+			Destination: filepath.Join("/EFI/systemd", bootConfig.systemdBootBinary),
+		})
 	}
 
 	err = systemBootPartitionMount.CleanClose()
+	if err != nil {
+		return nil, err
+	}
+
+	err = loopback.CleanClose()
+	if err != nil {
+		return nil, err
+	}
+
+	return outputArtifactsMetadata, nil
+}
+
+func replaceSuffix(input string, oldSuffix string, newSuffix string) string {
+	if !strings.HasSuffix(input, oldSuffix) {
+		return input
+	}
+	return strings.TrimSuffix(input, oldSuffix) + newSuffix
+}
+
+func writeInjectFilesYaml(metadata []OutputArtifactMetadata, baseConfigPath string) error {
+	type injectFileYaml struct {
+		InjectFiles []OutputArtifactMetadata `yaml:"injectFiles"`
+	}
+
+	yamlStruct := injectFileYaml{
+		InjectFiles: metadata,
+	}
+
+	// Marshal YAML
+	yamlBytes, err := yaml.Marshal(&yamlStruct)
+	if err != nil {
+		return fmt.Errorf("failed to marshal inject files metadata: %w", err)
+	}
+
+	// Construct YAML with header and inline unsigned comments
+	var builder strings.Builder
+	builder.WriteString("# This file is generated automatically by Prism output artifacts API.\n\n")
+
+	lines := strings.Split(string(yamlBytes), "\n")
+	i := 0
+	for _, item := range metadata {
+		for ; i < len(lines); i++ {
+			line := lines[i]
+			builder.WriteString(line + "\n")
+
+			// When you hit the 'source:' line, add the comment line
```

Comment:
This makes the output logic too complex. Let's just make the unsigned path a normal field in `OutputArtifactMetadata`. We can call it something like `unsignedSource`. We should ignore this field when inject-files is used.